File: find_eta.m
function eta = find_eta(ww, uu)
    flux_all_events = ww .* uu;
    flux_total = sum(flux_all_events, 'omitmissing');
    flux_downgradient_flag = flux_all_events .* flux_total > 0;
    eta = flux_total ./ sum(flux_all_events(flux_downgradient_flag), 'omitmissing');
end

File: utespac/find_eta.py
"""find_eta – transport efficiency η (total / downgradient flux ratio)."""

import numpy as np


def find_eta(ww: np.ndarray, uu: np.ndarray) -> float:
    """Compute η = total_flux / downgradient_flux.

    Parameters
    ----------
    ww : 1-D ndarray   w' perturbations.
    uu : 1-D ndarray   Scalar/velocity perturbations.

    Returns
    -------
    eta : float
    """
    flux_all   = ww * uu
    flux_total = np.nansum(flux_all)
    downgradient = (flux_all * flux_total) > 0
    dg_total = np.nansum(flux_all[downgradient])
    if dg_total == 0:
        return np.nan
    return flux_total / dg_total
